packages/test/test-end-to-end-tests/src/test/idCompressor.spec.ts

```diff
@@ -1084,3 +1085,91 @@ describeCompat("IdCompressor Summaries", "NoCompat", (getTestObjectProvider, com
 		assert(dds2.id.length > 8, "DDS's ID in attached container should not be short");
 	});
 });
+
+const dataStoreCount = 13;
+describeCompat("IdCompressor basic", "NoCompat", (getTestObjectProvider, apis) => {
+	const {
+		dataRuntime: { DataObject, DataObjectFactory },
+		containerRuntime: { ContainerRuntimeFactoryWithDefaultDataStore },
+	} = apis;
+	class TestDataObject extends DataObject {
+		public get _root() {
+			return this.root;
+		}
+
+		public get _context() {
+			return this.context;
+		}
+	}
+
+	const configProvider = createTestConfigProvider();
+	let provider: ITestObjectProvider;
+	const defaultFactory = new DataObjectFactory("T", TestDataObject, [], []);
+
+	const runtimeOptions: IContainerRuntimeOptions = {
+		enableRuntimeIdCompressor: "on",
+	};
+
+	const runtimeFactory = createContainerRuntimeFactoryWithDefaultDataStore(
+		ContainerRuntimeFactoryWithDefaultDataStore,
+		{
+			defaultFactory,
+			registryEntries: [[defaultFactory.type, Promise.resolve(defaultFactory)]],
+			runtimeOptions,
+		},
+	);
+
+	const createContainer = async (): Promise<IContainer> =>
+		provider.createDetachedContainer(runtimeFactory, { configProvider });
+
+	beforeEach("getTestObjectProvider", async function () {
+		provider = getTestObjectProvider();
+		// ODSP specific bug
+		if (provider.driver.type !== "odsp") {
+			this.skip();
+		}
+		configProvider.set("Fluid.ContainerRuntime.IdCompressorEnabled", true);
+		configProvider.set("Fluid.Runtime.UseShortIds", true);
+	});
+
+	function assertInvert(value: boolean, message: string) {
+		assert(!value, message);
+	}
+
+	itExpects(
+		"Id compressor bug repo when id is `[` which encodes to `%5B`",
+		[{ eventName: "fluid:telemetry:Container:ContainerClose", error: "No context for op" }],
+		async () => {
+			const container = await createContainer();
+			const dataObject = (await container.getEntryPoint()) as TestDataObject;
+			const containerRuntime = dataObject._context.containerRuntime;
+			for (let i = 0; i < dataStoreCount; i++) {
+				const createdObject = await defaultFactory.createInstance(containerRuntime);
+				dataObject._root.set(`${i}`, createdObject.handle);
+			}
+
+			await provider.attachDetachedContainer(container);
+
+			const handle = dataObject._root.get(`12`);
+			assert(handle !== undefined, "handle not found");
+			const childObject = await handle.get();
+			assert(childObject.id === "[", "This id is the problematic id");
+			childObject._root.set(`key13`, `value13`);
+			await provider.ensureSynchronized();
+
+			provider.resetDocumentServiceFactory();
+			const container2 = await provider.loadContainer(runtimeFactory, {
+				configProvider,
+			});
+
+			// This is the line that fails as we are getting the op that breaks the id
+			await provider.ensureSynchronized();
+
+			// This check isn't reached
+			assertInvert(
```

Comment:
You should also validate that it closes with "No context for op" error and that the address property is "[".

Reply:
Updated